packages/react/src/auto/AutoForm.ts

```diff
@@ -146,14 +149,19 @@ export const useAutoForm = <
   const isDeleteAction = metadata && isActionMetadata(metadata) && metadata.action.isDeleteAction;
   const isGlobalAction = action.type === "globalAction";
   const modelApiIdentifier = action.type == "action" ? action.modelApiIdentifier : undefined;
+  const isUpsertMetaAction = metadata && isActionMetadata(metadata) && fields.some((field) => field.metadata.fieldType === FieldType.Id);
+  const isUpsertWithFindBy = isUpsertMetaAction && !!findBy;
+
   const defaultValues: Record<string, unknown> = useMemo(
     () =>
       props.defaultValues ??
       (action.type === "globalAction"
         ? {}
         : {
-            [modelApiIdentifier!]: record ?? (!operatesWithRecordId && metadata && isActionMetadata(metadata) && metadata?.defaultRecord),
-            id: "0", // The ID value will be replaced when sending the form to use the record found by `findBy`
+            [modelApiIdentifier!]:
+              record ??
+              (!(operatesWithRecordId || isUpsertWithFindBy) && metadata && isActionMetadata(metadata) && metadata?.defaultRecord),
+            id: findBy,
```

Comment:
In the GQL variable maker, this will stomp out `modelApiId.id`. We used to send `id:"0" ` for create actions which was a lil strange. Like this, they are sent as undefined